src/Amazon.Common.DotNetCli.Tools/PosixUserHelper.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+
+namespace Amazon.Common.DotNetCli.Tools
+{
+    public struct PosixUserInfo
+    {
+        public int UserID;
+        public int GroupID;
+        public bool UserIDSet;
+        public bool GroupIDSet;
+    }
+
+    /// <summary>
+    /// Helper class to retrieve current Linux/Mac user and group information
+    /// </summary>
+    public static class PosixUserHelper
+    {
+        /// <summary>
+        /// Return True if running under Unix or Mac
+        /// </summary>
+        public static readonly bool IsRunningInPosix = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ||
+                                                       RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
```

Comment:
Hi - Norm had asked me to make the inclusion of using `System.Runtime.InteropServices` contingent upon NETCOREAPP3_1_OR_GREATER being set, so that this code won't bomb under Visual Studio.  This means that I can't add that as a parameter very easily (OSPlatform is not a c# enum but a "string-in-a-struct" kind of enum).

I was also thinking it wouldn't be a bad idea of separating the "Am I *NIX?" from operations like getting the ID, in case we wanted to add more functionality. 

What I am thinking I should do isto set up an injectable class-based implementation of launching a process and getting results, which would make this testable (we could mock `id` failing or otherwise not being on the system at all).  The same Process situation exists all over this code base, and this implementation could help make that code more testable as well.